client/src/components/AddChatView.tsx

```diff
@@ -1,11 +1,76 @@
-
+import  { ChangeEvent, useState, useContext } from 'react';
+import { UserContext } from './UserProvider';
+import ProfileImageUploader from './ProfileImageUploader';
+import useAddChat from '../hooks/useAddChat';
 
 const AddChatView = () => {
-    return (
-        <div>
-            AddChatView
-        </div>
-    );
-}
-
-export default AddChatView;
\ No newline at end of file
+  const userContext = useContext(UserContext);
+
+  if (!userContext) {
+    throw new Error('AddChatView must be used within a UserProvider');
```

Comment:
Nie rzucałbym błędu bo to wysypie całą aplikację. 
Nie mamy żadnego global error handlera, raczej warto sprawdzić co jest pod userContextem i np nie wyrenderować niczego albo wyświetlić tenże błąd

Reply:
więc teoretycznie ten sam approach co niżej z renderowaniem loadera,
można tutaj wrzucic useEffecta który sprawdza co z useContextem i jak jest załadowany to np zmienić wartość w nowym useState który odpowiada za "contextLoaded" -> i wtedy renderować resztę zamiast loadera